tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -5953,3 +5981,859 @@ def get_prod(arr):
+
+    cmplx128_from_float32 = uniform(low=min_float32 / 2, high=max_float32 / 2, size=size) + uniform(low=min_float32 / 2, high=max_float32 / 2, size=size) * 1j
+    # the result of the last operation is a Python complex type which has 8 bytes in the alignment,
+    # that's why we need to convert it to a numpy.complex64 the needed type.
+    cmplx64 = np.complex64(cmplx128_from_float32)
+    cmplx128 = uniform(low=min_float64 / 2, high=max_float64 / 2, size=size) + uniform(low=min_float64 / 2, high=max_float64 / 2, size=size) * 1j
+
+    f_bl = epyccel(get_chosen_elements, language=language)
+
+    assert (f_bl(bl) == get_chosen_elements(bl))
+
+    f_integer8 = epyccel(get_chosen_elements, language=language)
+    f_integer16 = epyccel(get_chosen_elements, language=language)
+    f_integer = epyccel(get_chosen_elements, language=language)
+    f_integer32 = epyccel(get_chosen_elements, language=language)
+
+    assert (f_integer8(integer8) == get_chosen_elements(integer8))
+    assert (f_integer16(integer16) == get_chosen_elements(integer16))
+    assert (f_integer(integer) == get_chosen_elements(integer))
+    assert (f_integer32(integer32) == get_chosen_elements(integer32))
+
+    f_integer64 = epyccel(get_chosen_elements, language=language)
+    assert (f_integer64(integer64) == get_chosen_elements(integer64))
+
+    f_fl = epyccel(get_chosen_elements, language=language)
+    f_fl32 = epyccel(get_chosen_elements, language=language)
+    f_fl64 = epyccel(get_chosen_elements, language=language)
+
+    assert (f_fl(fl) == get_chosen_elements(fl))
+    assert (f_fl32(fl32) == get_chosen_elements(fl32))
+    assert (f_fl64(fl64) == get_chosen_elements(fl64))
+
+    f_complex64 = epyccel(get_chosen_elements, language=language)
+    f_complex128 = epyccel(get_chosen_elements, language=language)
+
+    assert (f_complex64(cmplx64) == get_chosen_elements(cmplx64))
+    assert (f_complex128(cmplx128) == get_chosen_elements(cmplx128))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = [pytest.mark.fortran,
+            pytest.mark.skip(reason="Still under maintenance, See #770")]),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Needs a C printer see https://github.com/pyccel/pyccel/issues/791"),
+            pytest.mark.c]
+        ),
+        pytest.param("python", marks = [
+            pytest.mark.python,
+            pytest.mark.skip(reason="Missing Python printer")]
+        )
+    )
+)
+
+def test_numpy_where_array_like_1d(language):
+
+    @types('bool[:]')
+    @types('int[:]')
+    @types('int8[:]')
+    @types('int16[:]')
+    @types('int32[:]')
+    @types('int64[:]')
+    @types('float[:]')
+    @types('float32[:]')
+    @types('float64[:]')
+    @types('complex64[:]')
+    @types('complex128[:]')
+    def get_chosen_elements(arr):
+        from numpy import where, shape
+        a = where(arr)
+        s = shape(a)
+        return len(s), s[0], a[1], a[0]
```

Comment:
Assuming `where` is correctly implemented. you are returning an array here. E.g.:
```
In [6]: x = np.ones(4)

In [7]: a=np.where(x)

In [8]: a[0]
Out[8]: array([0, 1, 2, 3])
```